benchmarks/lib/src/jmh/java/com/amazon/corretto/crypto/provider/benchmarks/BenchmarkUtils.java

```diff
@@ -2,75 +2,38 @@
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+
 import java.security.Provider;
 import java.security.SecureRandom;
 import java.security.Security;
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.Set;
-
-import com.amazon.corretto.crypto.provider.AmazonCorrettoCryptoProvider;
-import org.bouncycastle.jce.provider.BouncyCastleProvider;
 
 class BenchmarkUtils {
-  private BenchmarkUtils() {}
+    private BenchmarkUtils() {
+    }
 
-  private static final SecureRandom sr = new SecureRandom();
-  private static final Set<String> NON_DEFAULT_PROVIDERS =
-      new HashSet(Arrays.asList("BC", "BCFIPS", "AmazonCorrettoCryptoProvider"));
-  private static final Provider[] DEFAULT_PROVIDERS;
+    private static final SecureRandom sr = new SecureRandom();
 
-  static {
-    DEFAULT_PROVIDERS = Security.getProviders();
-    for (Provider provider : DEFAULT_PROVIDERS) {
-      if (NON_DEFAULT_PROVIDERS.contains(provider.getName())) {
-        throw new RuntimeException("Provider prematurely (statically) registered: " + provider);
-      }
-    }
-  }
+    private static final Provider bc = new BouncyCastleProvider();
 
-  static byte[] getRandBytes(int n) {
-    byte[] ret = new byte[n];
-    final int bcMaxSize = 32768;
-    for (int ii = 0; ii < n; ii += bcMaxSize) {
-      byte[] data = new byte[bcMaxSize];
-      sr.nextBytes(data);
-      System.arraycopy(data, 0, ret, ii, Math.min(bcMaxSize, n - ii));
+    static byte[] getRandBytes(int n) {
+        final byte[] ret = new byte[n];
+        final int bcMaxSize = 32768;
+        final byte[] data = new byte[bcMaxSize];
+        for (int ii = 0; ii < n; ii += bcMaxSize) {
+            sr.nextBytes(data);
+            System.arraycopy(data, 0, ret, ii, Math.min(bcMaxSize, n - ii));
+        }
+        return ret;
     }
-    return ret;
-  }
 
-  static void setupProvider(String providerName) {
-    removeAllProviders();
-    switch (providerName) {
-      case "AmazonCorrettoCryptoProvider":
-        installDefaultProviders();
+    // In all the benchmarks, the getInstance methods of crypto services are invoked by passing
+    // the name of the algorithm and the provider that we are interested in. As a result, we
+    // just need to ensure that the provider has been installed.
+    static void setupProviders() {
         AmazonCorrettoCryptoProvider.install();
         AmazonCorrettoCryptoProvider.INSTANCE.assertHealthy();
-        break;
-      case "BC":
-        Security.insertProviderAt(new BouncyCastleProvider(), 1);
-        break;
-      case "SUN":
-      case "SunEC":
-      case "SunJCE":
-      case "SunRsaSign":
-        installDefaultProviders();
-        break;
-      default:
-        throw new RuntimeException("Unrecognized provider: " + providerName);
-    }
-  }
-
-  static void installDefaultProviders() {
-    for (Provider provider : DEFAULT_PROVIDERS) {
-      Security.addProvider(provider);
-    }
-  }
-
-  static void removeAllProviders() {
```

Comment:
I believe the original reason to remove/setup individual providers was to isolate utilities classes or other crypto instances that may be requested internally (_without_ specifying the provider explicitly) by providers. One instance of that can be found in [JDK11's MGF1](https://github.com/corretto/corretto-11/blob/1b08b745e538fb8f2c89d379382459622260f043/src/java.base/share/classes/sun/security/rsa/MGF1.java#L47), some others [here](https://github.com/search?q=repo%3Acorretto%2Fcorretto-11+%22.getInstance%22+path%3Asrc%2Fjava.base%2Fshare%2Fclasses%2F&type=code).

My concern here is that without isolating individual providers/provider suites, then we'll get less accurate measurements due to cross-provider instance sourcing. Do we need this particular change (installing all providers all the time) to satisfy the overall intent of the PR?

Reply:
Thanks for catching that.